iothub/device/src/IotHubBaseClient.cs

```diff
@@ -619,7 +641,9 @@ private protected static ClientPipelineBuilder BuildPipeline()
                 .With((ctx, innerHandler) => new RetryDelegatingHandler(ctx, innerHandler))
                 .With((ctx, innerHandler) => new ExceptionRemappingHandler(ctx, innerHandler))
                 .With((ctx, innerHandler) => new TransportDelegatingHandler(ctx, innerHandler))
-                .With((ctx, innerHandler) => transporthandlerFactory.Create(ctx));
+                .With((ctx, innerHandler) => transporthandlerFactory.Create(ctx, innerHandler))
+                // An HTTP layer is added for some operations that aren't available over MQTT or AMQP, including, e.g., file upload.
+                .With((ctx, innerHandler) => new HttpTransportHandler(ctx, innerHandler));
```

Comment:
The recovery was validated locally. 
It is not possible to leverage fault injection for these tests as the client "native" transport won't be http, so any open -> recovery attempt will be tied to mqtt/amqp layer and not http.
By the time the retriable http operation is executed, the client would have already recovered from the injected fault.